joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

diff_drive_controller:
  type: "diff_drive_controller/DiffDriveController"
  left_wheel: ["Left_forward_joint", "Left_backward_joint"]
  right_wheel: ["Right_forward_joint", "Right_backward_joint"]
  publish_rate: 50.0
  pose_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  wheel_separation_multiplier: 2.9 # default: 1.0
  wheel_radius_multiplier: 0.8 # default: 1.0
  cmd_vel_timeout: 0.2
  base_frame_id: base_footprint
  linear:
    x:
      has_velocity_limits: true
      max_velocity: 0.08 # m/s        <----------------------
      has_acceleration_limits: true
      max_acceleration: 0.6 # m/s^2   <----------------------
  angular:
    z:
      has_velocity_limits: true
      max_velocity: 0.14 # rad/s      <----------------------
      has_acceleration_limits: true
      max_acceleration: 1.0 # rad/s^2 <----------------------
  enable_odom_tf: true
  
  wheel_separation: 0.17
  wheel_radius: 0.034